### DISCUSSION TITLE
Unable to login github repository

### DISCUSSION THREAD

**Discussion by AL1073004** [community/community#27326] [Other Feature Feedback, Questions, & Ideas]
Hi Team, 

I am unable to login github repository because i have no service on mobile sim. Is there any other ways to login github repository . Please help me to resolve this issue

Thanks & Regards
Nagarani R

**Comment by airtower-luna** [comment]
If you have set up SMS as the second factor and can't login now, check the docs on [Recovering your account if you lose your 2FA credentials](https://docs.github.com/en/authentication/securing-your-account-with-two-factor-authentication-2fa/recovering-your-account-if-you-lose-your-2fa-credentials). I'm afraid the only easy way is if you have your recovery codes, but the article describes which alternatives exist.

If you want to set up two factor auth without needing a mobile connection (e.g. after recovering your account), use TOTP (with an authenticator app) instead of SMS. TOTP doesn't require any connection after setting it up (except to load the website, of course). And it's more secure anyway, because it can't be intercepted.
